### PR TITLE
Pullrequest apple push notifications aes encrypted caller message

### PR DESCRIPTION
**Pull request by BenSartor**
Hi @jritzerfeld,
I would recommend you review this pull request as a whole and not commit by commit.
Except for the last two commits. I pushed them to discuss the inspection warning with you. Maybe independ of this pull request via phone.
Cheers,
 Ben

### CODE REVIEW

File: src/main/java/org/simlar/simlarserver/utils/AesUtil.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            final Cipher cipher = Cipher.getInstance("AES/CBC/PKCS5PADDING");
+            cipher.init(mode, secretKeySpec, ivParameterSpec);
+            return cipher.doFinal(message);
+        } catch (final NoSuchAlgorithmException | NoSuchPaddingException | InvalidKeyException | InvalidAlgorithmParameterException | IllegalBlockSizeException | BadPaddingException e) {
+            throw new AesUtilException(ExceptionUtils.getRootCauseMessage(e), e);
+        }
+    }
+
+    @Nullable
+    public static String encrypt(final String message, final String initializationVector, final String password){
```
Comment (jritzerfeld) [inline src/main/java/org/simlar/simlarserver/utils/AesUtil.java]: Missing space before '{'

File: src/main/java/org/simlar/simlarserver/utils/AesUtil.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        return Base64.encodeBase64String(aes(Cipher.ENCRYPT_MODE, message.getBytes(StandardCharsets.UTF_8), initializationVector, password));
+    }
+
+    public static String decrypt(final String message, final String initializationVector, final String password){
```
Comment (jritzerfeld) [inline src/main/java/org/simlar/simlarserver/utils/AesUtil.java]: Missing space before '{'

File: src/main/java/org/simlar/simlarserver/utils/AesUtil.java (hunk 1 above)
Comment (BenSartor) [reply src/main/java/org/simlar/simlarserver/utils/AesUtil.java]: Fixed. Thank you!

File: src/main/java/org/simlar/simlarserver/utils/AesUtil.java (hunk 2 above)
Comment (BenSartor) [reply src/main/java/org/simlar/simlarserver/utils/AesUtil.java]: Fixed and thanks again!

**Pull request merged**